tax/combined_bracket.py

```diff
@@ -1,24 +1,28 @@
 """
-Generate combined brackets (state + federal) that are pre-adjusted by standard deduction
+Generate combined brackets (state + federal) that are pre-adjusted by
+standard deduction
 
 TODO: Figure out how to incorporate itemized deductions
 """
 
-combined = {}  # combined brackets go here. map state->filing_status->income_type
+# combined brackets go here. map state->filing_status->income_type
+combined = {}
 
 import bracket_definitions as bd
 import constants
 import deduction_definitions as dd
 
+
 def iter_bracket(bracket, reverse=False):
     """iterate through brackets in sorted order
     yield base_value, marginal_rate
     """
-    bracket_keys = sorted(bracket.iterkeys(), reverse=reverse)
+    bracket_keys = sorted(bracket.keys(), reverse=reverse)
```

Comment:
what's wrong with iterkeys?

Reply:
PY3 compat

Reply:
Of course, this changes the performance characteristic of this function, as python 3's .keys() IS python2's .iterkeys(). While here it really doesn't matter, I'm wondering how this works in general.   

Does py3 compatibility imply code should work in both python 2 and python 3, without the aid of a 2to3 tool?   Or must we be at the point where the code executes in both python 2 and python 3 and let python 2 take the (marginally) worse performance?

Reply:
> Does py3 compatibility imply code should work in both python 2 and python 3, without the aid of a 2to3 tool?

Yes, for the most part.  py3-only is also socially-acceptable at this point.  I haven't seen a 2to3 tool used in the wild in a long time.  Using `six` is pretty common.

> While here it really doesn't matter, I'm wondering how this works in general.

`six.iterkeys(d)` calls `d.iterkeys()` in py2 and `d.keys()` in py3, so that is a viable option.  I generally have small enough dicts that `d.keys()` is fine.